From: Thomas Bogendoerfer <tbogendoerfer@suse.de>
Subject: kabi mvpp2 10gkr support
Patch-Mainline: never, kABI
References: bsc#1098633

[js] PHY_INTERFACE_MODE_MAX is used only in code, not for
     declarations of arrays etc.

Signed-off-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/base/property.c |    2 ++
 include/linux/phy.h     |    2 ++
 include/linux/phy/phy.h |    2 ++
 3 files changed, 6 insertions(+)

--- a/drivers/base/property.c
+++ b/drivers/base/property.c
@@ -16,7 +16,9 @@
 #include <linux/of.h>
 #include <linux/of_address.h>
 #include <linux/of_graph.h>
+#ifndef __GENKSYMS__
 #include <linux/of_irq.h>
+#endif
 #include <linux/property.h>
 #include <linux/etherdevice.h>
 #include <linux/phy.h>
--- a/include/linux/phy.h
+++ b/include/linux/phy.h
@@ -84,9 +84,11 @@ typedef enum {
 	PHY_INTERFACE_MODE_1000BASEX,
 	PHY_INTERFACE_MODE_2500BASEX,
 	PHY_INTERFACE_MODE_RXAUI,
+#ifndef __GENKSYMS__
 	PHY_INTERFACE_MODE_XAUI,
 	/* 10GBASE-KR, XFI, SFI - single lane 10G Serdes */
 	PHY_INTERFACE_MODE_10GKR,
+#endif
 	PHY_INTERFACE_MODE_MAX,
 } phy_interface_t;
 
--- a/include/linux/phy/phy.h
+++ b/include/linux/phy/phy.h
@@ -27,8 +27,10 @@ enum phy_mode {
 	PHY_MODE_USB_HOST,
 	PHY_MODE_USB_DEVICE,
 	PHY_MODE_USB_OTG,
+#ifndef __GENKSYMS__
 	PHY_MODE_SGMII,
 	PHY_MODE_10GKR,
+#endif
 };
 
 /**
